assets/js/consent-mode/consent-mode.test.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Consent Mode tests
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import './consent-mode';
+
+describe( 'Consent Mode', () => {
+	const gtagMock = jest.fn();
+
+	beforeEach( () => {
+		global.gtag = gtagMock;
+		global._googlesitekitConsentCategoryMap = {
+			'test-category': [ 'test-parameter' ],
+		};
+	} );
+
+	afterEach( () => {
+		jest.restoreAllMocks();
+	} );
+
+	it( 'should call gtag with the correct parameters when wp_listen_for_consent_change event is triggered', () => {
+		document.dispatchEvent(
+			new CustomEvent( 'wp_listen_for_consent_change', {
+				detail: {
+					'test-category': 'allow',
+				},
+			} )
+		);
+
+		expect( gtagMock ).toHaveBeenCalledWith( 'consent', 'update', {
+			'test-parameter': 'granted',
+		} );
+	} );
+
+	it( 'should call gtag with the correct parameters when wp_consent_type_defined event is triggered', () => {
```

Comment:
This seems to be a duplicate, since the previous test is the same

Reply:
This test tests the route through the `wp_consent_type_defined` event:

https://github.com/google/site-kit-wp/blob/452957960539e9ae61275c93c6f2d2612a0f3c86/assets/js/consent-mode/consent-mode.js#L68-L71

Where as the first event is testing the route via the `wp_listen_for_consent_change` event:

https://github.com/google/site-kit-wp/blob/452957960539e9ae61275c93c6f2d2612a0f3c86/assets/js/consent-mode/consent-mode.js#L17-L42

Reply:
Ah I see, thanks